dist/api_config.py

```diff
@@ -193,12 +193,14 @@ def checkstr(c):
         return cfunc + ', NULL'
 
 def get_default(c):
+    cdef = c.flags
     t = gettype(c)
+    choices = cdef.get('choices', [])
     if c.default == 'false':
         return 'false'
     elif c.default == 'true':
         return 'true'
-    elif t == 'string' and c.default == 'none':
+    elif t == 'string' and c.default == 'none' and choices == []:
```

Comment:
This seems fine to me.

It could just be `not c.flags.get(choices, [])`, I think?